src/Shared/OSSGadget.cs

```diff
@@ -1,18 +1,110 @@
 {
     public class OSSGadget
     {
+        public OutputFormat currentOutputFormat = OutputFormat.text;
+
+        public OSSGadget()
+        {
+            CommonInitialization.Initialize();
+        }
+
         /// <summary>
-        /// Logger for this class
+        ///     Logger for this class
         /// </summary>
         public static NLog.ILogger Logger { get; set; } = NLog.LogManager.GetCurrentClassLogger();
 
-        public OSSGadget()
+        /// <summary>
+        ///     Formulates the help text for each derived tool
+        /// </summary>
+        /// <typeparam name="T"> </typeparam>
+        /// <param name="result"> </param>
+        /// <param name="errs"> </param>
+        protected void DisplayHelp<T>(ParserResult<T> result, IEnumerable<Error> errs)
         {
-            CommonInitialization.Initialize();
+            HelpText helpText = HelpText.AutoBuild(result, h =>
+            {
+                h.AddDashesToOption = true;
+                h.AutoVersion = true;
+                h.AdditionalNewLineAfterOption = false;
+                h.MaximumDisplayWidth = Console.WindowWidth;
+                h.AddPostOptionsLines(BaseProjectManager.GetCommonSupportedHelpTextLines());
+                return HelpText.DefaultParsingErrorsHandler(result, h);
+            });
+            Console.Error.Write(helpText);
+        }
+
+        /// <summary>
+        ///     Use the CommandlineParser library to get the cmd line arguments
+        /// </summary>
+        /// <typeparam name="T"> </typeparam>
+        /// <param name="args"> </param>
+        /// <returns> The Action object with the parsed options </returns>
+        protected ParserResult<T> ParseOptions<T>(string[]? args)
+        {
+            var parser = new Parser();
+            var parserResult = parser.ParseArguments<T>(args);
+            parserResult.WithNotParsed(errs => DisplayHelp(parserResult, errs));
+            return parserResult;
+        }
+
+        /// <summary>
+        ///     Restores the output stream to Console, if it was changed
+        /// </summary>
+        protected void RestoreOutput()
+        {
+            if (this.redirectConsole)
+            {
+                ConsoleHelper.RestoreConsole();
+            }
+        }
+
+        /// <summary>
+        ///     Use the OutputBuilder to select the given format and return a output builder
+        /// </summary>
+        /// <param name="format"> </param>
+        /// <returns> </returns>
+        protected IOutputBuilder SelectFormat(string? format)
+        {
+            try
+            {
+                currentOutputFormat = OutputBuilderFactory.GetOutputFormat(format ?? OutputFormat.text.ToString());
+                return OutputBuilderFactory.CreateOutputBuilder(this.currentOutputFormat);
+            }
+            catch (ArgumentOutOfRangeException)
+            {
+                Logger.Error("Invalid output format, selecting text");
```

Comment:
This doesn't warn the user that text is being selected if they pass null.

Reply:
resolving since you changed the function to non nullable argument

Reply:
added func doc